rundeckapp/grails-spa/packages/ui-trellis/src/app/components/readme-motd/main.ts

```diff
@@ -21,23 +22,38 @@ const i18nmessages = {
 const rundeckContext = getRundeckContext();
 
 rundeckContext.rootStore.ui.addItems([
-  {
-    section: "edit-project-file",
-    location: "main",
-    visible: true,
-    widget: markRaw(
-      defineComponent({
-        components: {
-          EditProjectFile,
-        },
-        inject: ["addUiMessages"],
-        props: ["itemData"],
-        created() {
-          this.addUiMessages([i18nmessages[locale]]);
-        },
-        template: `
-                <edit-project-file :filename="itemData.filename" :display-config="itemData.displayConfig" :project="itemData.project" :auth-admin="itemData.authAdmin"/>`,
-      }),
-    ),
-  },
-]);
+    {
+        section: 'edit-project-file',
+        location: 'main',
+        visible: true,
+        widget: markRaw(defineComponent({
+            components: {
+                EditProjectFile
+            },
+            props: ['itemData'],
+            inject: ['addUiMessages'],
+            //  configuring the code from here
+            data() {
+                return {
+                    // Initializing the data to be used in the component
+                    filename: '',
+                    displayConfig: [],
+                    project: '',
+                    authAdmin: false,
+                };
+            },
+            created() {
+                this.addUiMessages([i18nMessages[locale]]);
+                // Performing transformation on the data
+                this.filename = this.itemData.filename;
+                this.displayConfig = Array.isArray(this.itemData.displayConfig) ? this.itemData.displayConfig.filter((item: any) => typeof item === 'string') : [];
```

Comment:
Just a reminder to check:
- create 2 projects, one of them you should save a readme message, one you leave the readme message empty - and then ensure that they convert the displayConfig correctly, where none is converted to an empty array, and the one with a string is converted to an array with the string internally:

'[none]' -> []

'[projectName]' -> ["projectName"]

Reply:
I was reading the template of the EditProjectFile and realized that there is a scenario where it is relevant to have the array of strings with the content of "none". So the ideal solution should be:

'[none]' -> ["none"];
'[projectName]' -> ["projectName"];

the steps I'd like you to follow are:

- replace the square brackets from the string (so the string '[none]' will become 'none')
- then do a split on the string, splitting it by ", " (so the string 'none' will become an array ["none"])